### PR TITLE
Fixes #8921: Make the Rudder FusionInventory task work on Windows aga…

### PR DESCRIPTION
**Pull request by Kegeruneku**
…in and get the server roles

Ticket: https://www.rudder-project.org/redmine/issues/8921

### CODE REVIEW

File: rudder-agent/SOURCES/patches/fusioninventory/0020-correct-rudder-inventory-task.patch [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+         chomp $serverUuid;
+ 
+-        # get CFengine public key
++        # Get the CFengine public key
+         my $cfengineKey = getAllLines(
+             file => $cfengine_key_file
+         );
+ 
+-        # get owner name
++        # Get the current user name
```
Comment (Kegeruneku) [inline rudder-agent/SOURCES/patches/fusioninventory/0020-correct-rudder-inventory-task.patch]: Not really sure this is useful... (Fusion gives this already)

**Comment by ncharles** [comment]
<3

File: rudder-agent/SOURCES/patches/fusioninventory/0020-correct-rudder-inventory-task.patch [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+     SOFTWARES        => [ qw/COMMENTS FILESIZE FOLDER FROM HELPLINK INSTALLDATE
+                             NAME NO_REMOVE RELEASE_TYPE PUBLISHER
+
+--- fusioninventory-agent/lib/FusionInventory/Agent/Task/Inventory/Generic/Rudder.pm	2016-08-30 18:05:06.181949600 +0200
++++ fusioninventory-agent/lib/FusionInventory/Agent/Task/Inventory/Generic/Rudder.pm	2016-08-30 18:13:54.848011697 +0200
+@@ -7,8 +7,10 @@
+ 
+ use FusionInventory::Agent::Tools;
+ 
++my $uuid_hive = ($OSNAME eq 'MSWin32') ? 'C:\Program Files\Rudder\etc\uuid.hive' : '/opt/rudder/etc/uuid.hive';
```
Comment (peckpeck) [inline rudder-agent/SOURCES/patches/fusioninventory/0020-correct-rudder-inventory-task.patch]: You cannot be sure OSNAME is *defined* before this file is *loaded*.
You should put this little bit of login either where it is used or in a dedicated function.

File: rudder-agent/SOURCES/patches/fusioninventory/0020-correct-rudder-inventory-task.patch [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+@@ -17,23 +19,28 @@
+     my $inventory = $params{inventory};
+     my $logger    = $params{logger};
+ 
+-    # get Rudder UUID
+-    my $Uuid = getFirstLine(
+-        logger => $logger, file => '/opt/rudder/etc/uuid.hive'
++    # Get the Rudder UUID
++    my $uuid = getFirstLine(
++        logger => $logger, file => "$uuid_hive"
```
Comment (peckpeck) [inline rudder-agent/SOURCES/patches/fusioninventory/0020-correct-rudder-inventory-task.patch]: in perl variables don't need another indirection level to be protected. You can remove those ""

**Comment by peckpeck** [comment]
Aside of that, this looks good.

**Comment by ncharles** [comment]
Replaced by https://github.com/Normation/rudder-packages/pull/1062

**Pull request closed without merging**